src/structures/GuildMember.js

```diff
@@ -283,11 +283,11 @@ class GuildMember extends Base {
    */
   async edit(data, reason) {
     if (data.channel) {
-      data.channel = this.guild.channels.resolve(data.channel);
-      if (!data.channel || data.channel.type !== 'voice') {
+      let voiceChannelID = this.guild.channels.resolveID(data.channel);
```

Comment:
```suggestion
      const voiceChannelID = this.guild.channels.resolveID(data.channel);
```
should there also be some check if the channel is voice-based if its cached ?

Reply:
We could but these are some points that I had in my mind related to this:
1) The check will need another `resolve`, that might slow down the method a little bit.
2) Right now if the channel passed isn't a VC (cached or uncached), it will throw the same error that is returned by the API. While after adding the check, the errors will be **slightly** different depending on the channel passed was cached (lib error) or uncached (API error), which seems a little inconsistent from a user's perspective.

Reply:
wouldnt nessercerily require another resolve

something like

```js
const voiceChannel = this.guild.channels.cache.get(voiceChannelID);
if (voiceChannel && voiceChannel.type !== 'voice') { ... }```

Reply:
The second point still holds true tho. I feel like letting the API do the work here seems more consistent and also saves us from making that `get` call (which I know isn't that big of a deal but still 😅).

Reply:
Aight, It's 2-1 now, so I added your suggestion. Check it out and see if there is anything that I forgot :)👍